libparsec/crates/client/src/workspace/store/file_updater.rs

```diff
@@ -62,36 +64,48 @@ pub(super) async fn for_update_file(
             listener.await;
         }
 
-        let update_guard = {
-            let mut cache_guard = store.current_view_cache.lock().expect("Mutex is poisoned");
-
-            // 1) Lock for update
+        // 1) Lock for update
 
-            let outcome = cache_guard.lock_update_manifests.take(entry_id);
-            match outcome {
+        enum LockForUpdateOutcome {
+            WaitAndRetryStep1(EventListener),
+            GoToStep3(ManifestUpdateLockGuard),
+            GoToStep4((ManifestUpdateLockGuard, ArcLocalChildManifest)),
```

Comment:
```suggestion
            WaitAndRetryStep1(EventListener),
            // There is no GoToStep2 because it's the dispatching step
            GoToStep3(ManifestUpdateLockGuard),
            GoToStep4((ManifestUpdateLockGuard, ArcLocalChildManifest)),
```

Or something like that to clear up why there is no step 2 here

Reply:
:+1: